File: LeetCode001.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let map = new Map()
    for (let i in nums) {
        const j = map.get(target - nums[i])
        if (j) return [j, i]
        map.set(nums[i], i)
    }
};

File: LeetCode001.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    hash = Hash.new()
    nums.each.with_index do |n , i|
        j = hash[target - n]
        return [j, i] if j
        hash[n] = i
    end
end

puts two_sum([1,2,3], 5)